PRESUBMIT.py

```diff
@@ -54,6 +70,36 @@ def CheckPatchFormatted(input_api, output_api):
 
     is_format_required = git_cl_format_code == 2
 
+    if not is_format_required or input_api.PRESUBMIT_FIX:
+        # Use Prettier to format other file types.
+        prettier_types = (
+            '.js',
+            '.ts',
+            '.tsx',
+        )
+
+        files_to_format = [
+            f for f in _GetDiffFiles(input_api)
+            if os.path.isfile(f) and f.endswith(prettier_types)
+        ]
+
+        node_args = [
+            brave_node.PathInNodeModules('prettier', 'bin-prettier'),
+            '--write' if input_api.PRESUBMIT_FIX else '--check',
+        ]
+
+        files_per_command = 10 if input_api.is_windows else 1000
+        for i in range(0, len(files_to_format), files_per_command):
+            args = node_args + files_to_format[i:i + files_per_command]
+            try:
+                brave_node.RunNode(args)
+            except RuntimeError as err:
+                if 'Forgot to run Prettier?' in str(err):
```

Comment:
Can we use the exit code instead? Seems flakey to rely on the string, especially if they add translating or change it.

> The command will return exit code 1 in the second case, which is helpful inside the CI pipelines. 

https://prettier.io/docs/en/cli.html#--check

Reply:
I'll look into it.

Reply:
> Can we use the exit code instead? Seems flakey to rely on the string, especially if they add translating or change it.
> 
> > The command will return exit code 1 in the second case, which is helpful inside the CI pipelines.
> 
> https://prettier.io/docs/en/cli.html#--check

Node uses exit code `1` on many errors, including a missing file or any unhandled exception. Following this logic we will show "please run `npm run presubmit -- --fix`" in these cases, which will be confusing as it's not the real issue.

I'd like to avoid the string comparison, but I don't see any other way to distinguish the Prettier "format required" exit code from a Node "error occurred" exit code.

> Seems flakey to rely on the string, especially if they add translating or change it.

This should be handled during Prettier update in package.json on our side.

Reply:
Okay yeah, that makes sense. Just had a quick look at the source and they definitely aren't localizing it at the moment.

We'll have to keep this in mind when we do bump the prettier version though.